### ISSUE TITLE
Mod crashes clients on the fabric version (on a modpack)

### ISSUE THREAD

**Issue by Ilaskus** [ChiselsAndBits/Chisels-and-Bits#938]
Unsure whether this is an incompatibility or not, but whenever me or my friend walks up to a chest, our games crash. Could be a rendering issue, I don't know.
Client crash log:
[latest.log](https://github.com/ChiselsAndBits/Chisels-and-Bits/files/8941624/latest.log)
(Server logs indicate no problems with the mod whatsoever; it only shows us being disconnected.)

* MC Version: 1.18.2
* C&B Version: 1.2.90
* Do You have Optifine: No

**Comment by Ilaskus** [comment]
I think I see why now. Must've been something to do with WTHIT trying to render a chiseled block inside a chest onto its chest content preview, that's probably why it crashed my client and not the server. Sorry for the hassle.

**Comment by marchermans** [comment]
Hmm I thought I fixed that.... Seems like I did not.
I will see what I can do since that is clearly a bug when it tries to read the stack data from NBT.

**Comment by artman41** [comment]
@OrionDevelopment I've had the same crash except on forge - different Vanilla NBT call but the file location is the same.

Couldn't tell you what caused it; I was placing chiseled blocks when I and the other person on my server got booted with the same crash - server stayed up though.

**EDIT**: Just removed the mod and reloaded the server - looks like I might've triggered [Carry On](https://www.curseforge.com/minecraft/mc-mods/carry-on) to pickup the chisel block I was working on.

MC Version: 1.18.2
C&B Version: `forge 1.2.90` (Direwolf20 1.5.1 modpack)
Do You have Optifine: No

```log
---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 6/21/22, 8:24 PM
Description: Unexpected error

java.lang.NullPointerException: Cannot invoke "net.minecraft.nbt.ListTag.m_7264_()" because "nbt" is null
	at mod.chiselsandbits.block.entities.storage.SimpleStateEntryPalette.deserializeNBT(SimpleStateEntryPalette.java:59) ~[ChiselsBits-forge-1.18.2-1.2.90-universal.jar%2392!/:1.2.90] {re:classloading}
	at mod.chiselsandbits.block.entities.storage.SimpleStateEntryStorage.deserializeNBT(SimpleStateEntryStorage.java:306) ~[ChiselsBits-forge-1.18.2-1.2.90-universal.jar%2392!/:1.2.90] {re:classloading}
	at mod.chiselsandbits.block.entities.storage.SimpleStateEntryStorage.deserializeNBT(SimpleStateEntryStorage.java:30) ~[ChiselsBits-forge-1.18.2-1.2.90-universal.jar%2392!/:1.2.90] {re:classloading}
	...
```
[crash-2022-06-21_20.19.56-client.txt](https://github.com/ChiselsAndBits/Chisels-and-Bits/files/8952229/crash-2022-06-21_20.19.56-client.txt)
[crash-2022-06-21_20.24.19-client.txt](https://github.com/ChiselsAndBits/Chisels-and-Bits/files/8952230/crash-2022-06-21_20.24.19-client.txt)

**Comment by marchermans** [comment]
Thanks for all the reports, still fighting things in my real life, and with the damn shadows. But I am getting closer to a solution!

I will see if I can this fixed in the near future, all-though I wonder what is causing this to not have that exact subsection of the NBT.

**Comment by artman41** [comment]
@OrionDevelopment if you can monkey-patch a version that logs the NBT on the deserialisation, I can play with it for a while and see if it crashes.

Could be that some data in the NBT is corrupted somehow 🤷

**Comment by marchermans** [comment]
Fixed in 1.2.100
